### PR TITLE
59-BUNSPACE2024-Add-New-Styling-For-Background

### PR DESCRIPTION
**Pull request by jmorales8**
add switch night mode button and dark theme in the background, will be adding a moon or someshit in the back

## Extra Resources

- [The 10 commandments of navigating code reviews](https://techbeacon.com/app-dev-testing/10-commandments-navigating-code-reviews)
- [Google's Code Review Developer Guide](https://google.github.io/eng-practices/review/)

Closes [#59, #56, #57 ]

### CODE REVIEW

File: ui/src/components/NightModeButton/nightModeButton.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.night{
+  &__mode {
+    &__button {
+      width: 40px;
+      border-radius: 50%;
+    }
+  }
+}
```
Comment (GrpFruit) [inline ui/src/components/NightModeButton/nightModeButton.scss:8]: add empty line at the end

File: ui/src/components/AppContent/AppContent.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    throw new Error("DarkMode must be used within a ThemeProvider");
+  }
+  const { isDarkMode } = themeContext;
+  console.log(isDarkMode)
+  return(
+    <div className={isDarkMode ? "app__content__night" : "app"}>
+      {children}
+    </div>
+  )
+}
```
Comment (GrpFruit) [inline ui/src/components/AppContent/AppContent.tsx:15]: add empty line at the end

File: ui/src/Footer/Footer.tsx [hunk 3]
```diff
@@ -19,6 +20,9 @@ export function Footer() {
           </a>
         )
       })}
+      <div className="footer__button__night">
+        <NightModeButton />
+      </div>
     </footer>
   )
 }
```
Comment (GrpFruit) [inline ui/src/Footer/Footer.tsx:28]: add empty line at the end

**Comment by GrpFruit** [comment]
moral of the story add empty lines at the end of files
when switching tabs on the nav bar the night mode effect doesn't stay
also, the footer and nav bar don't change color either

**Comment by jmorales8** [comment]
> moral of the story add empty lines at the end of files when switching tabs on the nav bar the night mode effect doesn't stay also, the footer and nav bar don't change color either
> 
> edit: how did you get the dark mode color? try looking for a color palette that goes with the main theme so it makes sense visually
> 
> Screenshot 2024-08-05 at 11 21 54 AM

So the reason why this doesnt work currently is because the way our navbar works. Its currently using `<a>` tag links so when it goes to a new page, it refreshes and loads everything which is bad and resets the state of everything, including the music component and the theme too. In the music pr, we switched to react routers `<Links>` which DOES NOT refresh the next page and simply loads the component while keeping the state the same. This can be seen if you click the pillow icon at the top right which does use `<Link>`. In conclusion, it does work assuming i can get the pr for music in and we can get a new better navbar

As for the styling the other components have their own styling but there isnt a set color palete. I was just going off of what looks good.

**Comment by GrpFruit** [comment]
show her color palettes and have her choose the one she likes best, I bet its a burden for her to pick colors out of thin air and not knowing how they look after it's on the page. https://www.color-hex.com/color/eecbcb
I personally like this color for darkmode
Screenshot 2024-08-05 at 1 56 12 PM

**Comment by jmorales8** [comment]
> show her color palettes and have her choose the one she likes best, I bet its a burden for her to pick colors out of thin air and not knowing how they look after it's on the page. https://www.color-hex.com/color/eecbcb I personally like this color for darkmode Screenshot 2024-08-05 at 1 56 12 PM

I dont, imo. It looks rotten asf like as if it an actual bunny was mudered xd

**Comment by GrpFruit** [comment]
> > show her color palettes and have her choose the one she likes best, I bet its a burden for her to pick colors out of thin air and not knowing how they look after it's on the page. https://www.color-hex.com/color/eecbcb I personally like this color for darkmode Screenshot 2024-08-05 at 1 56 12 PM
> 
> I dont, imo. It looks rotten asf like as if it an actual bunny was mudered xd

I mean the purple is not bad if matched right with other like colors for a dark theme but ig the navbar and footer is throwing me off. Use a color palette once the main color is chosen for a dark theme. 

You have a ticket up for reworking the nav and footer right?

File: ui/src/Footer/Footer.tsx (hunk 3 above)
Comment (GrpFruit) [reply ui/src/Footer/Footer.tsx:28]: do this @jmorales8

File: ui/src/components/AppContent/AppContent.tsx (hunk 2 above)
Comment (GrpFruit) [reply ui/src/components/AppContent/AppContent.tsx:15]: and this @jmorales8

File: ui/src/components/NightModeButton/nightModeButton.scss (hunk 1 above)
Comment (GrpFruit) [reply ui/src/components/NightModeButton/nightModeButton.scss:8]: and this @jmorales8

**Review by GrpFruit — APPROVED** [review]
finish add empty lines in those files and this can be merged in

**Comment by GrpFruit** [comment]
> > > show her color palettes and have her choose the one she likes best, I bet its a burden for her to pick colors out of thin air and not knowing how they look after it's on the page. https://www.color-hex.com/color/eecbcb I personally like this color for darkmode Screenshot 2024-08-05 at 1 56 12 PM
> > 
> > 
> > I dont, imo. It looks rotten asf like as if it an actual bunny was mudered xd
> 
> I mean the purple is not bad if matched right with other like colors for a dark theme but ig the navbar and footer is throwing me off. Use a color palette once the main color is chosen for a dark theme.
> 
> You have a ticket up for reworking the nav and footer right?

also maybe try changing the scroll bar to dark mode too to see what it looks like. If it ain't funk Shuai change it back xD

**Comment by jmorales8** [comment]
> > > show her color palettes and have her choose the one she likes best, I bet its a burden for her to pick colors out of thin air and not knowing how they look after it's on the page. https://www.color-hex.com/color/eecbcb I personally like this color for darkmode Screenshot 2024-08-05 at 1 56 12 PM
> > 
> > 
> > I dont, imo. It looks rotten asf like as if it an actual bunny was mudered xd
> 
> I mean the purple is not bad if matched right with other like colors for a dark theme but ig the navbar and footer is throwing me off. Use a color palette once the main color is chosen for a dark theme.
> 
> You have a ticket up for reworking the nav and footer right?

yea but mainly the footer, i can make one for the footer if you want

**Pull request merged**